File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>


int calculateLetterCount(string text);
int calculateWordCount(string text);
int calculateSentenceCount(string text);
float calculateHundredsPerWord(int count, int wordCount);
float calculateIndex(float lettersPerHundredWords, float sentencesPerHundredWords);

int main(void)
{
    string text = get_string("Text: ");

    int letterCount = calculateLetterCount(text);
    int wordCount = calculateWordCount(text) + 1;
    int sentenceCount = calculateSentenceCount(text);

    float lettersPerHundredWords = calculateHundredsPerWord(letterCount, wordCount);
    float sentencesPerHundredWords = calculateHundredsPerWord(sentenceCount, wordCount);

    float index = calculateIndex(lettersPerHundredWords, sentencesPerHundredWords);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", index);
    }

}


int calculateLetterCount(string text)
{
    int letterCount = 0;
    int n = strlen(text);
    for (int i = 0; i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letterCount++;
        }
    }
    return letterCount;
}


int calculateWordCount(string text)
{
    int wordCount = 0;
    int n = strlen(text);
    for (int i = 0; i < n; i++)
    {
        if (isspace(text[i]))
        {
            wordCount++;
        }
    }
    return wordCount;
}

int calculateSentenceCount(string text)
{
    int sentenceCount = 0;
    int n = strlen(text);
    for (int i = 0; i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentenceCount++;
        }
    }
    return sentenceCount;
}

float calculateHundredsPerWord(int count, int wordCount)
{
    return ((float) count * 100) / (float) wordCount;
}

float calculateIndex(float lettersPerHundredWords, float sentencesPerHundredWords)
{
    return 0.0588 * lettersPerHundredWords - 0.296 * sentencesPerHundredWords - 15.8;
}

File: pset6/readability.py
from cs50 import get_string

text = get_string("Text: ")

letterCount = 0.0
wordCount = 1.0
sentenceCount = 0.0

for char in text:
    if char.isspace():
        wordCount += 1
    elif char == '.' or char == '!' or char == '?':
        sentenceCount += 1
    elif char.isalpha():
        letterCount += 1

lettersPerHundredWords = (letterCount * 100.0) / float(wordCount)
sentencesPerHundredWords =  (sentenceCount * 100.0) / float(wordCount)

index = 0.0588 * lettersPerHundredWords - 0.296 * sentencesPerHundredWords - 15.8

if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {round(index)}")

